### PR TITLE
adding twilio examples for node

### PR DESCRIPTION
**Pull request by nikolawhallon**
Why?

Customers have had trouble converting our Twilio Python examples to Node, so here is a basic integration.

Details:

Both `twilio-proxy-mono.js` and `twilio-proxy-stereo.js` empirically work, but I am not a JavaScript expert and I may be doing some things badly. In particular, working with arrays and buffers proved to be very frustrating, so I would like this work double checked.

These examples do not contain the buffering optimizations for throughput that the Python code has (i.e. these node examples will pass media messages along to Deepgram as soon as possible instead of buffering to send fewer, larger packets - this means these node examples effectively send 20ms chunks, which isn't as performant on throughout as, say, 100-400ms chunks). The reason I took this short cut was because, again, playing with JS buffers and arrays is not fun, but I can certainly add this back!

These examples also don't account for dropped Twilio/phone packets, which over the period of a very long call could result in slightly wrong timings across channels. This was again done to avoid too much massaging of those arrays and buffer, but I do think I could add this back.

On a more meta-note, is this even the right repository anymore ( @jviso , @MichaelJolley , @natalierutgers ) ? The (small amount of) code here does not seem well-maintained - for example, the Twilio Python example is slightly out of date (or at least the README is) - I'll make another PR to fix that up, but maybe this should live elsewhere. We now have this + blog posts + the repositories which go with blog posts + examples in our docs + examples in Missions in Console, and I think there could be some better organization of sample code - I'm always happy to contribute code, I however have no real ideas on how to organize it!

Either way, having this PR up for code review "somewhere" is certainly a good thing - will be requesting reviews from those who know node/JS better than I do.

### CODE REVIEW

File: twilio/node/twilio-proxy-stereo.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+const WebSocketServer = require('ws');
+const { Deepgram } = require('@deepgram/sdk')
+
+const websocketServer = new WebSocketServer.Server({ port: 5000 })
+const deepgram = new Deepgram("1eb45f404f76787f1eb8eb440505982264744776")
```
Comment (nikolawhallon) [inline twilio/node/twilio-proxy-stereo.js:5]:
```suggestion
const deepgram = new Deepgram("INSERT_YOUR_API_KEY_HERE")
```

Comment (nikolawhallon) [reply twilio/node/twilio-proxy-stereo.js]: I deleted this API Key so it can't be used anymore anyways!

File: twilio/node/twilio-proxy-mono.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  deepgramLive.addListener('transcriptReceived', (transcription) => {
+    console.dir(transcription, { depth: null });
+  });
+
+  ws.on("message", data => {
+    var twilio_message = JSON.parse(data);
+    if (twilio_message["event"] === "connected" || twilio_message["event"] === "start") {
+      console.log("received a twilio connected or start event");
+    }
+    if (twilio_message["event"] === "media") {
```
Comment (mlodato517) [inline twilio/node/twilio-proxy-mono.js:25]: This  can be an `else if` it looks :shrug:

File: twilio/node/twilio-proxy-mono.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      var media = twilio_message["media"];
+      var audio = Buffer.from(media["payload"], "base64");
+      if (deepgramLive && deepgramLive.getReadyState() === 1) {
```
Comment (mlodato517) [inline twilio/node/twilio-proxy-mono.js:28]: I don't know if we're targeting a specific JS spec but I remember there being stuff like `deepgramLive?.getReadyState() === 1` for this common idiom.

File: twilio/node/twilio-proxy-mono.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  });
+
+  ws.on("close", () => {
+    console.log("client has connected");
+    if (deepgramLive && deepgramLive.getReadyState() === 1) {
+      deepgramLive.finish();
+    }
+  });
+
+  ws.onerror = function () {
```
Comment (mlodato517) [inline twilio/node/twilio-proxy-mono.js:41]: Is there a reason this one is the only one defined this way? That is, `function ()` instead of `() =>`?

**Review by mlodato517 — COMMENTED** [review]
I don't remember JS at all but I dropped some random things I _think_ I remember. Seemed like they applied to both files so I only left them on one. Hopefully someone who actually JS's will come to help :see_no_evil:

File: twilio/node/twilio-proxy-mono.js (hunk 4 above)
Comment (nikolawhallon) [reply twilio/node/twilio-proxy-mono.js:41]: I dunno, I have no clue what I'm doing tbh

**Pull request merged**